addressbook/interface/exchangeAutoCompleteResult/mivExchangeAutoCompleteResult.js

```diff
@@ -247,53 +244,52 @@ mivExchangeAutoCompleteResult.prototype = {
      * persistent storage as well.
      */
     //void removeValueAt(in long rowIndex, in boolean removeFromDb);
-    removeValueAt: function _removeValueAt(aRowIndex, removeFromDb) {},
+    removeValueAt: function _removeValueAt(aRowIndex, removeFromDb) {
+        var localid = this._idcards.splice(aRowIndex, 1);
+        delete this._cards[localid];
+    },
 
     //void addResult(in mivExchangeAbCard aCard);
     addResult: function _addResult(aCard) {
         // First check if this card is not already in the list
         var cardExists = false;
-        if (this._cards) {
-            for (var card of Object.values(this._cards)) {
-                if (card.localId == aCard.localId) {
 
-                    cardExists = true;
-                    break;
-                }
-            }
+        if (this._cards
+            && this._cards[aCard.localId]) {
+                cardExists = true;
```

Comment:
You could simply `return` instead of using a variable + another `if` condition.

Reply:
Well seen, thanks.